### ISSUE TITLE
Version of Kubernetes

### ISSUE THREAD

**Issue by holymoly** [zeebe-io/zeebe-helm#6]
Hello, 

I'm facing an issue when istalling the chart. Im running kubeadm 1.16.3 and the api from Deployments seems to have changed from:
```
apiVersion: extensions/v1beta1
kind: Deployment
```
to
```
apiVersion: apps/v1
kind: Deployment
```

as described here:
https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

After changing and testing I got now an Error telling me:
```
Error: release zeebefull failed: Deployment.apps "zeebefull-nginx-ingress-default-backend" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"nginx-ingress", "component":"default-backend", "release":"zeebefull"}: `selector` does not match template `labels`
```
I gues it is related to the apps/v1 change. Some googling brought me to:https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates

Its stating " plan your selectors up front". I'm not 100% in the helm templates. Could it be that we assign the labels dynamically?

Kind Regards,
holymoly

**Comment by salaboy** [comment]
@holymoly can you check which version of the chart are you actually using? 
Also, in which cloud provider are you running>?

**Comment by holymoly** [comment]
I used the zeebe-full-0.0.19.tgz chart unpack and modified as describes.

I'm running it on-prem in my home lab and build the setup on kubeadm 1.16.3 and use Metallb (h ttps://metallb.universe.tf/) as a loadbalancer. So no cloudprovider. 
Its a mix of linux host with arm and amd64 architechture

**Comment by salaboy** [comment]
@holymoly what do you mean that you unpack it? Are you not using ‘helm install’ ? With the version from the helm repo?

**Comment by holymoly** [comment]
i tried with the repo:
```
helm install --name zeebe-full zeebe/zeebe-full
```
where i got the error:
```
Error: validation failed: unable to recognize "": no matches for kind "Deployment" in version "extensions/v1beta1"
```
So I cloned this repo and unpacked zeebe-full-0.0.19.tgz. Then i modified  
```
apiVersion: extensions/v1beta1
kind: Deployment
```
to
```
apiVersion: apps/v1
kind: Deployment
```
to fit my kubadm 1.16.3.
Then i run 
```
 helm install --name zeebe-full  zeebe-full
```
where zeebe-full is the folder that contains the modified files. Which executes with the error:
```
Error: release zeebe-full failed: Deployment.apps "zeebe-full-nginx-ingress-controller" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"nginx-ingress", "component":"controller", "release":"zeebe-full"}: `selector` does not match template `labels`]
```

Hope this makes it a bit clearer. I know on-prem is not so usual for kubernetes, but i like to test things locally to have full control before using a cloud provider.

**Comment by salaboy** [comment]
@holymoly that is because there is a chart dependency that also has that difference.. can you send a PR with your changes?

**Comment by holymoly** [comment]
@salaboy you want a PR with the error?
```
Error: release zeebe-full failed: Deployment.apps "zeebe-full-nginx-ingress-controller" is invalid: [spec.selector: Required value, spec.template.metadata.labels: Invalid value: map[string]string{"app":"nginx-ingress", "component":"controller", "release":"zeebe-full"}: `selector` does not match template `labels`]
```
If so I can do in a view hours.

**Comment by salaboy** [comment]
@holymoly you can send a PR with the fix.. to the follow repos. You know what the error is.. you also know how to fix it, from your previous comments. 
So the error is basically stating that now you also require to add the labels in the spec -> selector.. 
Here are the repos with the source code of the charts:
- https://github.com/zeebe-io/zeebe-full-helm
- https://github.com/zeebe-io/zeebe-cluster-helm
- https://github.com/zeebe-io/zeebe-operate-helm

**Comment by holymoly** [comment]
@salaboy I fixed the issue on the api stuff and the label. When I look in to the repos you mentioned  i cannot find anything defining that. Are the repos missing some folders?
Also I just started working with helm 3 Weeks ago so I would like to understand how to fix that but with this repos I dont know how to be honested. 

Just for info, I could fix the issue changing controller-deployment.yaml and default-backend-deployemnt.yaml directly in the nginx-ingress folder, but as stated I can not find the source in the repos you mentinoned.

**Comment by salaboy** [comment]
@holymoly the repos that I've linked are the Zeebe ones, which add dependencies via the requirements.yaml files for example for ingress: https://github.com/zeebe-io/zeebe-full-helm/blob/master/zeebe-full/requirements.yaml

That means that in the `values.yaml` file we can customise the labels of the nginx-ingress-controller .. We might also want to upgrade the nginx-ingress controller dependency version to make sure that one works with Kubernetes 1.16

**Comment by salaboy** [comment]
Ok solved in Zeebe Full Chart 0.0.21 -> https://github.com/zeebe-io/zeebe-full-helm/releases/tag/v0.0.21
